react/src/components/HeaderMenu/HeaderMenuContainer.tsx

```diff
@@ -98,44 +97,65 @@ const HeaderMenuContainer = () => {
   const [boxesDataForBulkOperation, setBoxesDataForBulkOperation] = useState<IBoxDetailsData[]>([]);
 
   const onScanningDone = useCallback(
-    (qrResolvedValues: QrResolvedValue[]) => {
-      if (qrResolvedValues.length === 1) {
-        const singleResolvedQrValue = qrResolvedValues[0];
+    (IQrResolvedValues: IQrResolvedValue[]) => {
+      if (IQrResolvedValues.length === 1) {
+        const singleResolvedQrValue = IQrResolvedValues[0];
         switch (singleResolvedQrValue.kind) {
-          case "success": {
-            const boxLabelIdentifier = singleResolvedQrValue.value.labelIdentifier;
+          case QrResolverResultKind.SUCCESS: {
+            const boxLabelIdentifier = singleResolvedQrValue?.value.labelIdentifier;
             navigate(`/bases/${baseId}/boxes/${boxLabelIdentifier}`);
             break;
           }
-          case "noBoxtributeQr": {
-            toast({
-              title: `Scanned QR code is not a Boxtribute QR code`,
-              status: "error",
-              isClosable: true,
-              duration: 2000,
+          case QrResolverResultKind.NOT_BOXTRIBUTE_QR: {
+            createToast({
+              title: "Error",
+              type: "error",
+              message: "Error: Scanned QR code is not a Boxtribute QR code",
             });
             break;
           }
-          case "notAssignedToBox": {
-            toast({
-              title: `Scanned QR code is not assigned to a box yet`,
-              status: "info",
-              isClosable: true,
-              duration: 2000,
+          case QrResolverResultKind.NOT_AUTHORIZED: {
+            createToast({
+              title: "Error",
+              type: "error",
+              message: "Error: You don't have access to the box assigned to this QR code",
             });
-            // TODO: uncomment this once we have finished/tested the Create Box feature sufficiently
-            // navigate(
-            //   `/bases/${baseId}/boxes/create?qrCode=${singleResolvedQrValue.qrCodeValue}`
-            // );
+            break;
+          }
+          case QrResolverResultKind.LABEL_NOT_FOUND: {
+            createToast({
+              title: "Error",
+              type: "error",
+              message: "Error: Box not found for this label",
+            });
+            break;
+          }
+          case QrResolverResultKind.FAIL: {
+            createToast({
+              title: "QR Reader",
+              type: "error",
+              message: `Error - Code ${singleResolvedQrValue?.error.code}: Cannot retrieve data for the QR code`,
+            });
+            break;
+          }
+          case QrResolverResultKind.NOT_ASSIGNED_TO_BOX: {
+            createToast({
+              title: "QR Code",
+              type: "info",
+              message: "Scanned QR code is not assigned to a box yet",
+            });
+
+            navigate(`/bases/${baseId}/boxes/create?qrCode=${singleResolvedQrValue?.qrCodeValue}`);
```

Comment:
@vahidbazzaz the route has changed, it's `.../create/${singleResolvedQrValue?.qrCodeValue}`

Reply:
@pylipp good catch. I'll change it to the new route.